### DISCUSSION TITLE
Serverside issue while using Web3Wallet in NextJS

### DISCUSSION THREAD

**Discussion by 0xLPircy** [WalletConnect/walletconnect-monorepo#3765] [Developer Support] [answered]
I installed using 
 `npm install @walletconnect/web3wallet`

Added to my layout.js:
``` jsx
import { Core } from "@walletconnect/core";
import { Web3Wallet } from "@walletconnect/web3wallet";

const core = new Core({
  projectId: process.env.PROJECT_ID,
});

const web3wallet = await Web3Wallet.init({
  core, // <- pass the shared `core` instance
  metadata: {
    name: "Demo app",
    description: "Demo Client as Wallet/Peer",
    url: "www.walletconnect.com",
    icons: [],
  },
});
```

I am receiving error:
` Server Error
Error: To use WalletConnect server side, you'll need to install the "lokijs" dependency. If you are seeing this error during a build / in an SSR environment, you can add "lokijs" as a devDependency to make this error go away.
`

**Comment by glitch-txs** [comment, marked as answer]
you can run init function inside a useEffect to avoid it being called in the server, but this error can be ignored as well.
